### ISSUE TITLE
Disambiguation not loading

### ISSUE THREAD

**Issue by xferguson** [esterpantaleo/etymology#21]
this started after my last pull from master. When I do a search for a term, I get the following in my console:

`````
1280
load.js:182 loading languages
etytree.js:25 searching word in database
etytree.js:30 https://etytree-virtuoso.wmflabs.org/sparql?query=PREFIX%20dbetym%3A%20%3Ch…s.org%2F%2Fdbnaryetymology%23EtymologyEntry%3E%20.%20%20%20%20%20%7D%20%7D
dagre.js:174 Object {}

d3.v3.min.js:1 Error: <g> attribute transform: Expected number, "translate(Infinity,20)scal…".
i @ d3.v3.min.js:1
(anonymous) @ d3.v3.min.js:3
Y @ d3.v3.min.js:1
Co.each @ d3.v3.min.js:3
Co.attr @ d3.v3.min.js:3
(anonymous) @ dagre.js:201
t @ d3.v3.min.js:1
(anonymous) @ d3.v3.min.js:1
c @ d3.v3.min.js:3
(anonymous) @ d3.v3.min.js:3
(anonymous) @ d3.v3.min.js:3
Y @ d3.v3.min.js:1
Co.each @ d3.v3.min.js:3
n.event @ d3.v3.min.js:3
drawDisambiguationDAGRE @ dagre.js:246
source.subscribe.response @ etytree.js:34
a.__tryOrUnsub @ Rx.min.js:43
a.next @ Rx.min.js:42
a._next @ Rx.min.js:40
a.next @ Rx.min.js:40
req.onload @ load.js:12

etytree.js:36 done disambiguation
`````

I used line breaks to separate out the error from the rest of the code. I will look at the new changes and see what I discover.

**Comment by xferguson** [comment]
This is, in fact, related to the code-cleanup ( pull request #18 ) and the changing of `==` to `===`

dagre.js, line 170 was `if (iris == "") {`, which was returning true, even though `iris` was an object with no values in it( `[""]` ).

So, in order to check type properly, I changed the line to `if (iris.keys.length === 0) {`

This should help keep other issues from cropping up.

Now the disambiguation is loading, but the normal etymology tree isn't loading.

I get "Sorry, the server cannot extract etymological relationships correctly for this word." on the front end, and the following error in the console:

`````
done DAGRE2
dagre.js:326 done DAGREZIP

Rx.min.js:43 Uncaught TypeError: Cannot read property 'eqIri' of undefined
    at dagre.js:351
    at Array.forEach (<anonymous>)
    at drawData (dagre.js:338)
    at a._next (dagre.js:320)
    at a.__tryOrUnsub (Rx.min.js:43)
    at a.next (Rx.min.js:42)
    at a._next (Rx.min.js:40)
    at a.next (Rx.min.js:40)
    at a._next (Rx.min.js:40)
    at a.next (Rx.min.js:40)
(anonymous) @ dagre.js:351
drawData @ dagre.js:338
(anonymous) @ dagre.js:320
a.__tryOrUnsub @ Rx.min.js:43
a.next @ Rx.min.js:42
a._next @ Rx.min.js:40
a.next @ Rx.min.js:40
a._next @ Rx.min.js:40
a.next @ Rx.min.js:40
a.checkIterators @ Rx.min.js:102
a.notifyNext @ Rx.min.js:105
a._next @ Rx.min.js:61
a.next @ Rx.min.js:40
req.onload @ load.js:22
`````

I suspect it is from a similar "type" error. This is all on the cleanup, branch, and I will let you know when they are all resolved.

**Comment by xferguson** [comment]
The tree draws fine now. The drawData function in dagre.js was indeed the same problem. there were several `node[element.xxx.xxx] == null` type checks, that failed as `=== null` because they were actually type `undefined`

I added an or clause to get `undefined === node[element.xxx.xxx] || null === node[element.xxx.xxx]` and it is working. All fixed code is in pull request #18
